autopush/endpoint.py

```diff
@@ -200,3 +220,14 @@ def _finish_missed_store(self, result=None):
         self.metrics.increment("router.broadcast.miss")
         self.write("Success")
         self.finish()
+
+    def write_error(self, code, exception=None):
+        """ Write the error (otherwise unhandled exception when dealing with
+        unknown protocol specifications.) """
+        reason = "No reason"
+        if exception is not None and exception.reason is not None:
+            reason = exception.reason
+        error = "%d: %s" % (code, reason)
+        self.set_status(code)
+        log.err("Endpoint write_error: %s" % error)
```

Comment:
log.err can take a plain exception, it'll properly wrap it, and then raven will properly extract all the exception details for reporting